### PR TITLE
WLS Debug

### PR DESCRIPTION
**Pull request by dewagter**
WLS Debug

### CODE REVIEW

**Comment by gautierhattenberger** [comment]
Is this working when WLS is used at several places (e.g. stab and guidance) ?

**Comment by tmldeponti** [comment]
> Is this working when WLS is used at several places (e.g. stab and guidance) ?

That is a good question. I made this changes having in mind my oneloop controller which runs the WLS only once. Maybe @dewagter or @fvantienen have an idea on what would happen/ how to best adapt the changes for the nested stab guid control loops.

**Comment by tmldeponti** [comment]
@gautierhattenberger @fvantienen @dewagter  Indeed the original structure did not allow for the WLS to be run twice in the same control loop. To solve this issue we propose to:

1. Provide the WLS struct, containing all the parameters, as an input to the functions in wls_alloc.c (streaming message functions and the main function)
2. Create wrapper functions to call the streaming message function from the control modules where the wls allocation is used (e.g oneloop, stabilization and guidance). When calling we can specify the loop name (https://github.com/tudelft/paparazzi/commit/1dfbf1d20ce990fe54c0e2d0c2069c1e66b7283c#diff-2c51c778025662f50737e091542126267e0142aebf3e1f558f6eea5290420cd6R436) 
3. The declaration of the WLS struct  type stays in the the header file of the WLS but its definition moves to the target control module. This way multiple structs can be defined. 
4. I made it more clear that WLS_N_U and WLS_N_V are not simply the number of actuators and number of control axis. They are the maximum number of actuators and axis that will be supplied to the wls_alloc. So for example in the case of the nested stabilization and guidance, WLS_N_U and WLS_N_V should be the maximum values between the two calls to allocate enough memory to all variables. I do not think this was made super clear before.

I have already tested the changes with the oneloop controller in simulation (see https://github.com/tudelft/paparazzi/tree/rw3c_merging_v2). If you have suggestions let me know, then I can armonize all the other WLS calls in paparazzi.

**Comment by tmldeponti** [comment]
tested in sim on RW3F with stab and guid INDI. Drone flies. Messages seem to be streamed only for the stabilization and not the guidance.

**Comment by tmldeponti** [comment]
Checked a log and the system works. Both WLs stab and guid are being logged.

**Comment by tmldeponti** [comment]
Done

File: sw/airborne/math/wls/wls_alloc.c [hunk 1]
```diff
@@ -90,102 +112,99 @@ static void qr_solve_wrapper(int m, int n, float** A, float* b, float* x) {
...
- * @param gamma_sq Preference of satisfying control objective over desired
- * control vector (sqare root of gamma)
+ * @param gamma_sq Preference of satisfying control objective over desired control vector (sqare root of gamma)
  * @param imax Max number of iterations
- * @param n_u Length of u (the number of actuators)
- * @param n_v Lenght of v (the number of control objectives)
- *
+ * @param nu Length of u (the number of actuators)
+ * @param nv Length of v (the number of control objectives)
+ * @param WLS_p Struct that contains most of the WLS parameters
```
[the remark this answers is not in the document]
Comment (tmldeponti) [reply sw/airborne/math/wls/wls_alloc.c]: done

**Pull request merged**